docs/agented-vm-guide.md

```diff
@@ -0,0 +1,121 @@
+# Agented VM User Guide
```

Comment:
Add Table of Contents

Reply:
Other docs all have introduction after toc, should we follow a similar pattern?